java/client/src/main/java/glide/managers/ConnectionManager.java

```diff
@@ -122,6 +123,14 @@ private ConnectionRequest.Builder setupConnectionRequestBuilderBaseConfiguration
             connectionRequestBuilder.setInflightRequestsLimit(configuration.getInflightRequestsLimit());
         }
 
+        if (configuration.getReadFrom() == ReadFrom.AZ_AFFINITY) {
+            if (configuration.getClientAZ() != null) {
```

Comment:
consider flipping the conditions and removing the `else` (think: avoid nesting; fail early)
```
        if (configuration.getReadFrom() == ReadFrom.AZ_AFFINITY) {
            if (configuration.getClientAZ() == null) {
                throw new ConfigurationError("client_az must be set when read_from is set to AZ_AFFINITY");
            }
            connectionRequestBuilder.setClientAz(configuration.getClientAZ());
        }


```